tests/collapse-perf.rs

```diff
@@ -156,8 +156,8 @@ collapse_perf_tests! {
 
 fn test_collapse_perf(test_file: &str, expected_file: &str, options: Options) -> io::Result<()> {
     let r = BufReader::new(File::open(test_file)?);
-
-    let mut result = Cursor::new(Vec::with_capacity(expected_file.len()));
+    let expected_len = fs::metadata(expected_file)?.len() as usize;
+    let mut result = Cursor::new(Vec::with_capacity(expected_len));
```

Comment:
That's a neat trick! I wonder if we also want to play the same trick in `bin/collapse-perf.rs`?

Reply:
Sure, and maybe cap the buffer size to `128 * 1024` in case the file is really large?

```Rust
let len = fs::metadata(f)?.len() as usize;
let r = BufReader::with_capacity(std::cmp::min(len, 128 * 1024), File::open(f)?);
handle_file(options, r, io::stdout().lock())
```

Reply:
Actually, in `bin/collapse-perf.rs` don't we need to have the entire file in memory at once? Maybe I'm misremembering? One alternative here would be to just [mmap](https://docs.rs/memmap/) the entire file :sweat_smile:

Reply:
In flamegraph we read the input files into memory completely so we can sort the lines, but don't we only read the file a line at a time in collapse-perf?

Reply:
Ah, yes, you're right. I had my thinking reversed!

Reply:
So do you want to leave those as is with `BufReader`s, or should we use `mmap`?

Reply:
I'd leave them as `BufReader`s for now. `mmap` comes with its own set of `unsafe` restrictions that I don't think are worth it.